packages/i18n/async-locale-data/async-locale-data.js

```diff
@@ -33,18 +39,38 @@ class AsyncLocaleData extends React.Component {
     }
   }
 
-  loadLocaleData = locale => {
+  loadLocaleData = async locale => {
     const language = extractLanguageFromLocale(locale);
-    loadI18n(language).then(
-      data => {
-        this.setState({
-          isLoading: false,
-          language,
-          messages: data,
-        });
-      },
-      error => reportErrorToSentry(error, {})
-    );
+
+    let applicationMessagePromise = null;
+
+    if (typeof this.props.applicationMessages === 'function') {
+      applicationMessagePromise = this.props.applicationMessages(language);
+    }
+
+    const messageFetchingPromises = do {
+      if (applicationMessagePromise) {
+        [loadI18n(language), applicationMessagePromise];
+      } else {
+        [loadI18n(language)];
+      }
+    };
+
+    try {
+      const [messages, applicationMessages] = await Promise.all(
+        messageFetchingPromises
+      );
+      this.setState({
+        isLoading: false,
+        language,
+        messages: mergeMessages(
+          messages,
+          applicationMessages || this.props.applicationMessages[language]
```

Comment:
Use `this.props.applicationMessages && this.props.applicationMessages[language]` since the prop is optional and could run into runtime errors

Reply:
👍

Reply:
applicationMessages is defined as being a required prop in the docs though, and the current propType is object.isRequired? Should we change it to optional?